src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -123,11 +123,23 @@ internal static int Start(
                 throw new ConsoleHostStartupException(ConsoleHostStrings.ShellCannotBeStartedWithConfigConflict);
             }
 
-            // put PSHOME in front of PATH so that calling `powershell` within `powershell` always starts the same running version
+            // Put PSHOME in front of PATH so that calling `pwsh` within `pwsh` always starts the same running version.
             string path = Environment.GetEnvironmentVariable("PATH");
-            string pshome = Utils.DefaultPowerShellAppBase + Path.PathSeparator;
+            string dotnetToolsPathSegment = Path.Combine(".dotnet", "tools", ".store");
+            string pshome = Utils.DefaultPowerShellAppBase;
 
-            // to not impact startup perf, we don't remove duplicates, but we avoid adding a duplicate to the front
+            int index = pshome.IndexOf(dotnetToolsPathSegment, StringComparison.Ordinal);
+            if (index > 0)
+            {
+                // We're running PowerShell global tool. In this case the real entry executable should be the 'pwsh'
+                // or 'pwsh.exe' within the '../.dotnet/tools' folder, not what PSHome is pointing to.
+                // BTW, 13 is the length of '.dotnet/tools'
+                pshome = pshome[0..(index + 13)];
```

Comment:
this is true for global tools that are installed globally with `dotnet tool install -g`. It can be installed to any path, but I agree that most people will install it globally. Maybe add this to comments.

Reply:
Good point, thanks! I've updated the code to cover the scenario where the PowerShell global tool is installed to an arbitrary location.